page-tracking/src/page_tracker.rs

```diff
@@ -553,12 +554,17 @@ impl HypPageAlloc {
         }
     }
 
-    /// Takes and cleans `count` contiguous Pages with the requested alignment from the system map.
-    /// Sets the hypervisor as the owner of the pages, and any pages consumed up until that point,
-    /// in the system page map. Allows passing ranges of pages around without a mutable
-    /// reference to the global owners list. Panics if there are not `count` pages available. The
-    /// returned pages are eligible to be mapped into the host's address space.
-    pub fn take_pages(&mut self, count: usize, align: u64) -> SequentialPages<ConvertedClean> {
+    // Core allocator function. Finds `count` contiguous pages with the requested alignment from the
+    // system map. Sets the hypervisor as the owner and `page_state` as the state, for the returned
+    // pages and for any pages consumed for alignment purposes.
+    // Returns a pair of supervisor addresses containing the first and the last page address of the
+    // allocated range.
+    fn alloc_pages(
+        &mut self,
+        count: usize,
+        align: u64,
+        page_state: PageState,
+    ) -> (SupervisorPageAddr, SupervisorPageAddr) {
```

Comment:
Use `SupervisorPageRange` instead of a pair of addresses?

Reply:
Oh, much better. Yes!

Reply:
Updated. I thought about using `SupervisorPageRange` for the assignment iteration as well, but the iterator is destructive and I would've had to create two.